app/models/ability.rb

```diff
@@ -48,14 +48,15 @@ def initialize(user)
       can [:create], AgreementLetter
       can [:new, :create], Request
       cannot :view_personal_details, ApplicationLetter, user: { id: !user.id }
+      can [:show, :index, :archive], Event
+      cannot [:edit, :update, :new, :create, :destroy, :accept_all_applicants, :print_applications_eating_habits, :participants_pdf, :participants], Event
     end
     if user.role? :coach
       # Coaches can view Applications and participants for and view, upload and download materials for Event
       can [:view_applicants, :view_participants, :view_material, :upload_material, :print_applications, :download_material], Event
       can [:view_and_add_notes, :show], ApplicationLetter
-      can [:print_applications], Event
       can [:show, :index], Request
-      cannot :view_apply_button, Event
+      cannot [:view_apply_button, :edit, :update, :new, :create, :destroy], Event
```

Comment:
Again - only `cannot [:view_apply_button]` is needed here, because it revokes the access given to guests at line 36.